packages/synchro-charts/src/components/sc-dial/utils/fontWidth.ts

```diff
@@ -0,0 +1,43 @@
+const DEFAULT_FONT_SIZE = 12;
+const DEFAULT_FONT_FAMILY = 'sans-serif';
+
+/**
+ * Gets the width coefficient for each character
+ */
+const getTextWidthMap = (mapStr: string): Record<string, number> => {
+  const map: Record<string, number> = {};
+  if (typeof JSON === 'undefined') {
+    return map;
+  }
+  for (let i = 0; i < mapStr.length; i += 1) {
+    const char = String.fromCharCode(i + 32);
+    const size = (mapStr.charCodeAt(i) - 20) / 100;
+    map[char] = size;
+  }
+  return map;
+};
+
+/**
+ * Gets the approximate length of a string of characters
+ *
+ * Use "007LLmW'55;N0500LLLLLLLLLL00NNNLzWW\\\\WQb\\0FWLg\\bWb\\WQ\\WrWWQ000CL5LLFLL0LL**F*gLLLL5F0LF\\FFF5.5N"
+ * to gets an array of width coefficients for 95 common characters
```

Comment:
Changed to get the length of the label by `el in

Reply:
I don't understand your reply, could you elaborate?

Reply:
The purpose of this function is to get the length of a word. And one of the requirements in `sc-dial-svg` component is to get the length of the label to automatically determine the x value of the label and icon at the center of the dial. 

Now change to `componentDidLoad` in the life cycle in `sc-dial-svg` component, that is, after Render, get the length and y coordinates of the label, then re-determine the x coordinates of the label and the x and y coordinates of the icon, and re-render, so that the label and icon will be in the center of the dial